opentelekomcloud/resource_opentelekomcloud_compute_instance_v2.go

```diff
@@ -484,6 +489,14 @@ func resourceComputeInstanceV2Create(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
+	if hasFilledOpt(d, "auto_recovery") {
```

Comment:
Instead of creating the extra method, in Go you can just do an assignment and check inside the if:
```golang
if v, ok := d.GetOkExists("auto_recovery"); ok {
  ar := v.(bool)
  ...
}
```

Reply:
You are right under this scene which there is only one condition expression for 'if'. However if there are two or more condition expressions for 'if', such as 'if ** && ** ', then the wrapping method is  clearer. See the example[1]. So how about keeping it as it is?
[1] https://github.com/terraform-providers/terraform-provider-opentelekomcloud/pull/18/commits/aa77ebee4f8fc694a0d241d62d6b20aae5fe5bd9#diff-fa734f8e7b6638dfc32604a5e18322fdR259

Reply:
I admit that function preference is a subjective thing, so if you prefer it that is fine. In my opinion its not the typical way it would be approached in Go. I think Go very much prefers the verbose forms of things for clarity and repetition over terseness and dependency, etc. But I won't hold up the review on a preference.

---

Regarding your referenced commit, usage of reflection (`reflect` package) is definitely frowned upon. Reflection is not clear, and I get why you would look to it for a provider to clean up some of this boilerplate, but that is not really how Go is written. Go favors verbosity over terse/clever code, that's why you write `if err != nil` over and over and over. It's not the way providers are written, they manually should map all the fields, as it’s so different to the other providers - we’d be unable to accept a PR with reflection used to map the resource's schema.

Here are some thoughts from Rob Pike, one of the languages' creators, about reflections use cases (taken from his [Go Proverbs](https://go-proverbs.github.io/)): https://www.youtube.com/watch?v=PAAkCSZUG1c&t=14m35s

I'd be happy to discuss this further in the Slack room or over email.